### DISCUSSION TITLE
Can node open app windows and interact with apps cross platforms? Including doing text input.

### DISCUSSION THREAD

**Discussion by jsoneaday** [nodejs/help#4649] [Q&A]
I would like to create a node based script that opens and closes app windows and also inputs text and clicks buttons. But I need to do this across platforms.
Can node alone do this or do I need something like Cypress or Puppeteer?

**Comment by atlowChemi** [comment]
Puppeteer is written with Nodejs, but it uses the [CDP](https://chromedevtools.github.io/devtools-protocol/) to interact with Chrome.
You could technically implement the CDP calls directly with Node, but most likely that would not be worth the investment
